sdk/core/azure-core/inc/azure/core/internal/http/user_agent_helper.hpp

```diff
@@ -35,7 +35,7 @@ namespace Azure { namespace Core { namespace Http { namespace _internal {
      * the SDK code for compatibility reasons.
      *
      */
-    static std::string GenerateUserAgent(
+    static std::string BuildUserAgent(
```

Comment:
Since we are returning it, Generate was better, or even `Get`, if you want it shorter/less redundant and  following SDK precedence).

Build seems off (makes you think there is some builder pattern happening when it isn't).

Reply:
IMHO, as the architect: Get is inappropriate because Get implies that you are retrieving a value that is already calculated (which this explicitly is not). 

We do have an existing pattern of Create for factory constructors (Attestation uses this pattern to avoid requiring a "magic method").

Reply:
That's a good point about "Get".

Are you suggesting we use Create over Generate? That seems better than Build.

Reply:
I agree `get` is not a good choice since in C++. In C++ get implies the function does not modify the state of the object. Get methods are more often than not `const`

Reply:
We are keeping the `Generate` now.